libs/sql-parser/src/main/java/io/crate/sql/tree/IndexDefinition.java

```diff
@@ -40,6 +43,12 @@ public IndexDefinition(String ident, String method, List<T> columns, GenericProp
         this.method = method;
         this.columns = columns;
         this.properties = properties;
+        Set<T> uniqueColumns = new HashSet<>(columns());
+        if (uniqueColumns.size() != columns.size()) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ENGLISH, "Index %s contains duplicate columns.", ident)
+            );
+        }
```

Comment:
I think we should validate anything in the analysis phase which cannot be ensured by the grammer, such at `InnerTableElementsAnalyzer`.
Also the added test uses the analyzer scope, otherwise it should be a parser test.